todo/core.py

```diff
@@ -0,0 +1,342 @@
+            - `todo` The todo task
+        """
+        data = await self.cache.get_user_data(ctx.author.id)
+        pinned = pinned is not None and pinned is True
+        payload = {"task": todo, "pinned": pinned}
+        data["todos"].append(payload)
+        await self.cache.set_user_item(ctx.author, "todos", data["todos"])
+
+        data = data["user_settings"]
+        msg = f"Added that as a todo."
+        if data["extra_details"]:
+            msg += f"\n'{todo}'"
+        if data["use_timestamps"]:
+            msg += f"\n{timestamp_format()}"
+        if len(msg) > 2000:
+            await ctx.send_interactive(pagify(msg))
+        else:
+            await ctx.send(msg)
+        await self._maybe_autosort(ctx.author)
+
+    @todo.command(name="list")
+    async def todo_list(self, ctx: commands.Context):
+        """List your todos
+
+        This will list them with pinned todos first and then whatever sorting you have
+        """
+        data = await self.cache.get_user_data(ctx.author.id)
+        todos = data["todos"]
+        completed = data["completed"]
+        user_settings = data["user_settings"]
+
+        if not todos and not all([completed, user_settings["combine_lists"]]):
+            return await ctx.send(self._no_todo_message.format(prefix=ctx.clean_prefix))
+        elif not todos:
+            return await self.__complete_list(completed, **user_settings)
+
+        pinned, todos = await self._get_todos(todos)
+        todos = await formatting._format_todos(pinned, todos, **user_settings)
+        if completed and user_settings["combine_lists"]:
+            completed = await formatting._format_completed(
+                completed, combined=True, **user_settings
+            )
+            todos.extend(completed)
+
+        await self.page_logic(ctx, todos, f"{ctx.author.name}'s Todos", **user_settings)
+
+    @todo.command(name="multiadd")
+    async def todo_multi_add(self, ctx: commands.Context, *, todos: str = None):
+        """Add multiple todos in one command!"""
+        if ctx.message.reference and not any(
+            [todos is not None, ctx.message.attachments]
+        ):
+            # Message references get checked first
+            msg = ctx.message.reference.resolved
+            if not msg.attachments:
+                return await ctx.send("That message does not have files!")
+            maybe_file = msg.attachments[0]
+            if not maybe_file.filename.endswith(".txt"):
+                return await ctx.send("File format must be `.txt`")
+            todos = await maybe_file.read()
+            todos = todos.decode()
+        elif ctx.message.attachments:
+            maybe_file = ctx.message.attachments[0]
+            if not maybe_file.filename.endswith(".txt"):
+                return await ctx.send("File format must be `.txt`")
+            todos = await maybe_file.read()
+            todos = todos.decode()
```

Comment:
```suggestion
        if ctx.message.reference and not any(
            [todos is not None, ctx.message.attachments]
        ):
            # Message references get checked first
            msg = ctx.message.reference.resolved
            if not msg.attachments:
                return await ctx.send("That message does not have files!")
            maybe_file = msg.attachments[0]
            if not maybe_file.filename.endswith(".txt"):
                return await ctx.send("File format must be `.txt`")
        elif ctx.message.attachments:
            maybe_file = ctx.message.attachments[0]
            if not maybe_file.filename.endswith(".txt"):
                return await ctx.send("File format must be `.txt`")
        todos = await maybe_file.read()
        todos = todos.decode()
```

Github was really awkward on me to review this, but it should be an appropriate change

Reply:
The reason I do this is because `todos` can be a string that the user inputs running the command (for example, `[p]todo multiadd testing
Do something else`). It's messy, but it doesn't raise any unnecessary issues